### ISSUE TITLE
[BUG] GitHub download fails due to incorrect download URL

### ISSUE THREAD

**Issue by BastianZim** [conda-incubator/grayskull#267]
**Describe the bug**

`grayskull pypi --strict-conda-forge https://github.com/psi4/psi4` fails with missing URL.

**To Reproduce**
1. Run `grayskull pypi --strict-conda-forge https://github.com/psi4/psi4`
2. Get error: `Exception: 404 Client Error: Not Found for url: https://codeload.github.com/psi4/psi4/tar.gz/v1.5%2C%202021-11-27`

**Expected behavior**

The correct URL should be used.

**Outputs**
If applicable, add the output to help explain your problem.

Full log:

```zsh
#### Initializing recipe for https://github.com/psi4/psi4 ####

Starting the download of the sdist package psi4
Package seems to be missing on pypi.
Exception: 404 Client Error: Not Found for url: https://codeload.github.com/psi4/psi4/tar.gz/v1.5%2C%202021-11-27
```

**Environment:**
 - conda list
```shell
# packages in environment at /usr/local/Caskroom/miniforge/base/envs/grayskull:
#
# Name                    Version                   Build  Channel
alabaster                 0.7.12                     py_0    conda-forge
babel                     2.9.1              pyh44b312d_0    conda-forge
brotlipy                  0.7.0           py310he24745e_1003    conda-forge
bzip2                     1.0.8                h0d85af4_4    conda-forge
c-ares                    1.18.1               h0d85af4_0    conda-forge
ca-certificates           2021.10.8            h033912b_0    conda-forge
certifi                   2021.10.8       py310h2ec42d9_1    conda-forge
cffi                      1.15.0          py310hcc37b68_0    conda-forge
charset-normalizer        2.0.8              pyhd8ed1ab_0    conda-forge
colorama                  0.4.4              pyh9f0ad1d_0    conda-forge
cryptography              36.0.0          py310ha82f1d4_0    conda-forge
curl                      7.80.0               hf45b732_0    conda-forge
docutils                  0.17.1          py310h2ec42d9_1    conda-forge
expat                     2.4.1                he49afe7_0    conda-forge
gettext                   0.19.8.1          hd1a6beb_1008    conda-forge
git                       2.34.1          pl5321h9a53687_0    conda-forge
grayskull                 0.9.1              pyhd8ed1ab_0    conda-forge
idna                      3.1                pyhd3deb0d_0    conda-forge
imagesize                 1.3.0              pyhd8ed1ab_0    conda-forge
jinja2                    3.0.3              pyhd8ed1ab_0    conda-forge
krb5                      1.19.2               hcfbf3a7_3    conda-forge
libblas                   3.9.0           12_osx64_openblas    conda-forge
libcblas                  3.9.0           12_osx64_openblas    conda-forge
libcurl                   7.80.0               hf45b732_0    conda-forge
libcxx                    12.0.1               habf9029_0    conda-forge
libedit                   3.1.20191231         h0678c8f_2    conda-forge
libev                     4.33                 haf1e3a3_1    conda-forge
libffi                    3.4.2                h0d85af4_5    conda-forge
libgfortran               5.0.0           9_3_0_h6c81a4c_23    conda-forge
libgfortran5              9.3.0               h6c81a4c_23    conda-forge
libiconv                  1.16                 haf1e3a3_0    conda-forge
liblapack                 3.9.0           12_osx64_openblas    conda-forge
libnghttp2                1.43.0               h6f36284_1    conda-forge
libopenblas               0.3.18          openmp_h3351f45_0    conda-forge
libssh2                   1.10.0               h52ee1ee_2    conda-forge
libzlib                   1.2.11            h9173be1_1013    conda-forge
llvm-openmp               12.0.1               hda6cdc1_1    conda-forge
markupsafe                2.0.1           py310he24745e_1    conda-forge
ncurses                   6.2                  h2e338ed_4    conda-forge
numpy                     1.21.4          py310hd5a656b_0    conda-forge
openssl                   1.1.1l               h0d85af4_0    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pcre2                     10.37                ha16e1b2_0    conda-forge
perl                      5.32.1          1_h0d85af4_perl5    conda-forge
pip                       21.3.1             pyhd8ed1ab_0    conda-forge
progressbar2              3.53.1             pyh9f0ad1d_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pygments                  2.10.0             pyhd8ed1ab_0    conda-forge
pyopenssl                 21.0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.6              pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1           py310h2ec42d9_4    conda-forge
python                    3.10.0          h1248fe1_3_cpython    conda-forge
python-utils              2.5.6              pyh44b312d_0    conda-forge
python_abi                3.10                    2_cp310    conda-forge
pytz                      2021.3             pyhd8ed1ab_0    conda-forge
rapidfuzz                 1.8.3           py310hdd25497_0    conda-forge
readline                  8.1                  h05e3726_0    conda-forge
requests                  2.26.0             pyhd8ed1ab_1    conda-forge
ruamel.yaml               0.17.17         py310he24745e_1    conda-forge
ruamel.yaml.clib          0.2.6           py310he24745e_0    conda-forge
ruamel.yaml.jinja2        0.2.4                      py_1    conda-forge
setuptools                59.4.0          py310h2ec42d9_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snowballstemmer           2.2.0              pyhd8ed1ab_0    conda-forge
sphinx                    4.3.1              pyh6c4a22f_0    conda-forge
sphinxcontrib-applehelp   1.0.2                      py_0    conda-forge
sphinxcontrib-devhelp     1.0.2                      py_0    conda-forge
sphinxcontrib-htmlhelp    2.0.0              pyhd8ed1ab_0    conda-forge
sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
sphinxcontrib-qthelp      1.0.3                      py_0    conda-forge
sphinxcontrib-serializinghtml 1.1.5              pyhd8ed1ab_1    conda-forge
sqlite                    3.37.0               h23a322b_0    conda-forge
stdlib-list               0.7.0                      py_2    conda-forge
tk                        8.6.11               h5dbffcc_1    conda-forge
tzdata                    2021e                he74cb21_0    conda-forge
urllib3                   1.26.7             pyhd8ed1ab_0    conda-forge
wheel                     0.37.0             pyhd8ed1ab_1    conda-forge
xz                        5.2.5                haf1e3a3_1    conda-forge
zlib                      1.2.11            h9173be1_1013    conda-forge
```
 - pip list
```zsh
Package                       Version
----------------------------- ---------
alabaster                     0.7.12
Babel                         2.9.1
brotlipy                      0.7.0
certifi                       2021.10.8
cffi                          1.15.0
charset-normalizer            2.0.8
colorama                      0.4.4
cryptography                  36.0.0
docutils                      0.17.1
grayskull                     0.9.1
idna                          3.1
imagesize                     1.3.0
Jinja2                        3.0.3
MarkupSafe                    2.0.1
numpy                         1.21.4
packaging                     21.3
pip                           21.3.1
progressbar2                  3.53.1
pycparser                     2.21
Pygments                      2.10.0
pyOpenSSL                     21.0.0
pyparsing                     3.0.6
PySocks                       1.7.1
python-utils                  2.5.6
pytz                          2021.3
rapidfuzz                     1.8.3
requests                      2.26.0
ruamel.yaml                   0.17.17
ruamel.yaml.clib              0.2.6
ruamel.yaml.jinja2            0.2.4
setuptools                    59.4.0
six                           1.16.0
snowballstemmer               2.2.0
Sphinx                        4.3.1
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
stdlib-list                   0.7.0
urllib3                       1.26.7
wheel                         0.37.0
```
 - grayskull --version
```zsh
0.9.1
```
**Additional context**

**Comment by jaimergp** [comment]
Looks like it's trying to use the [release](https://github.com/psi4/psi4/releases/tag/v1.5) _name_ instead of the corresponding tag.

**Comment by BastianZim** [comment]
Hmm makes sense. That would be these lines, right?
https://github.com/conda-incubator/grayskull/blob/b012b7f41846c9eb4240db27797ebb358ceb8c52/grayskull/pypi/pypi.py#L81-L101

**Comment by jan00s** [comment]
Hey,

I faced the same issue and fixed this locally. Please have a look at this PR: https://github.com/conda-incubator/grayskull/pull/271.

**Comment by marcelotrevisani** [comment]
Fixed by #271 
Thanks @jan00s 
I will generate a new version in a few days
